modules/oas-page-builder/src/services/pageBuilder.ts

```diff
@@ -50,39 +65,76 @@ const getAtlasSpecUrl = async (apiKeyword: string) => {
+  output,
+  apiVersion,
+  resourceVersion,
+}: GetOASpecParams) {
+  try {
+    let spec = '';
+    const buildOptions: RedocBuildOptions = {};
+
+    if (sourceType === 'url') {
+      spec = source;
+    } else if (sourceType === 'local') {
+      const localFilePath = normalizePath(`${repoPath}/source/${source}`);
+      spec = localFilePath;
+    } else if (sourceType === 'atlas') {
+      spec = await getAtlasSpecUrl({ apiKeyword: source, apiVersion, resourceVersion });
+      // Ignore "incompatible types" warnings for Atlas Admin API/cloud-docs
+      buildOptions['ignoreIncompatibleTypes'] = true;
+    } else {
+      throw new Error(`Unsupported source type "${sourceType}" for ${pageSlug}`);
+    }
+
+    const finalFilename = normalizePath(`${output}/${pageSlug}/index.html`);
+    await redocExecutor.execute(spec, finalFilename, buildOptions);
+  } catch (e) {
+    console.error(e);
+  }
+}
+
 export const buildOpenAPIPages = async (
   entries: [string, OASPageMetadata][],
   { output, redoc: redocPath, repo: repoPath, siteUrl, siteTitle }: PageBuilderOptions
 ) => {
   const redocExecutor = new RedocExecutor(redocPath, siteUrl, siteTitle);
 
   for (const [pageSlug, data] of entries) {
-    const { source_type: sourceType, source } = data;
-
-    try {
-      let spec = '';
-      const buildOptions: RedocBuildOptions = {};
-
-      if (sourceType === 'url') {
-        spec = source;
-      } else if (sourceType === 'local') {
-        const localFilePath = normalizePath(`${repoPath}/source/${source}`);
-        spec = localFilePath;
-      } else if (sourceType === 'atlas') {
-        spec = await getAtlasSpecUrl(source);
-        // Ignore "incompatible types" warnings for Atlas Admin API/cloud-docs
-        buildOptions['ignoreIncompatibleTypes'] = true;
-      } else {
-        throw new Error(`Unsupported source type "${sourceType}" for ${pageSlug}`);
-      }
+    const { source_type: sourceType, source, api_version: apiVersion, resource_versions: resourceVersions } = data;
 
-      const finalFilename = normalizePath(`${output}/${pageSlug}/index.html`);
-      await redocExecutor.execute(spec, finalFilename, buildOptions);
-    } catch (e) {
-      console.error(e);
-      // Continue to try to build other pages since it's possible that mut will
-      // still upload existing HTML files
+    if (!apiVersion && resourceVersions && resourceVersions.length > 0) {
+      console.error(
+        `ERROR: API version is not specified, but resource version is present for source ${source} and sourceType: ${sourceType}`
+      );
       continue;
     }
+
+    if (resourceVersions) {
+      // if a resource versions array is provided, then we can loop through the resourceVersions array and call the getOASpec
+      // for each minor version
+      for (const resourceVersion of resourceVersions) {
+        await getOASpec({ source, sourceType, output, pageSlug, redocExecutor, repoPath, apiVersion, resourceVersion });
+      }
```

Comment:
Note that if there are resource versions for an API version, we should also create a page for the general API version. This page will have all of the resource versions bundled together into one, while the individual resource version pages are specific to that version. The tables in [this section](https://docs.google.com/document/d/1F2tmPnFM1Kj_gFZDAyWJUAO0M5L1Bd6xQ4pHTcmaI9M/edit#heading=h.k7miji4vp0u2) have example file outputs!

Reply:
Gotcha! Apologies for missing this, will update